src/librustc_typeck/check/demand.rs

```diff
@@ -395,16 +393,21 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         // `ExprKind::DropTemps` is semantically irrelevant for these suggestions.
         let expr = expr.peel_drop_temps();
 
+        let remove_prefix = |s: String, prefix: &str| {
```

Comment:
I think you can simplify some of the use sites further if you make this a `replace_prefix` function that has a further string argument to prepend after removing the prefix.

Reply:
Ah no, I see now, this won't work for the actual use cases inserting derefs

Reply:
Actually that convenient! I added it. :D